typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -155,14 +171,19 @@ async function sendMessage(
   const value = await gasCalc.estimatePaymentForMessage(expected);
 
   log('Sending message', { origin, destination });
+  let sent = false;
 
   await new Promise<ethers.ContractReceipt[]>((resolve, reject) => {
-    setTimeout(
-      () => reject(new Error('Timeout waiting for message receipt')),
-      MESSAGE_RECEIPT_TIMEOUT,
-    );
+    setTimeout(() => {
+      if (sent) {
+        reject(new Error('Timeout waiting for message receipt'));
+      } else {
+        reject(new Error('Timeout attempting to send message'));
+      }
+    }, MESSAGE_RECEIPT_TIMEOUT);
     app
       .sendHelloWorld(origin, destination, msg, value, (receipt) => {
+        sent = true;
```

Comment:
should we expose separate hooks? this `sent` usage seems brittle
have you tested this?

Reply:
I plan to test before we deploy, since it is only being used to give a better error message the consequence of it not working is somewhat low. I guess if we wanted a hook for when message is received too? Generally the whole thing kinda feels janky with the send waiting for receipt as well.